### PR TITLE
sword: 1.8.1 -> 1.9.0

### PR DESCRIPTION
**Pull request by afh**
## Description of changes

* Update to 1.9.0
* Migrate to `pkgs/by-name`
* Remove superfluous use of `pname`
* Improve `src.url` path by using `lib.version.majorMinor`
* Migrate to `finalAttrs` and `hash` from `rec` and `sha256`
* Add `dev` output for include files
* Remove `with lib` from `meta`
* Change `homepage` url scheme to `https`
* Ran nixfmt

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sw/sword/package.nix [hunk 1]
```diff
@@ -1,32 +1,53 @@
...
+  ];
 
   prePatch = ''
     patchShebangs .;
   '';
 
-  configureFlags = [ "--without-conf" "--enable-tests=no" ];
+  configureFlags = [
+    "--without-conf"
+    "--enable-tests=no"
```
Comment (AndersonTorres) [inline pkgs/by-name/sw/sword/package.nix:38]:
```suggestion
    (lib.withFeature true "conf")
    (lib.enableFeature false "tests")
```

File: pkgs/by-name/sw/sword/package.nix [hunk 2]
```diff
@@ -1,32 +1,53 @@
...
+  ];
+
   CXXFLAGS = [
```
Comment (AndersonTorres) [inline pkgs/by-name/sw/sword/package.nix:41]:
```suggestion
  env.CXXFLAGS = toString [
```

File: pkgs/by-name/sw/sword/package.nix [hunk 3]
```diff
@@ -36,9 +57,8 @@ stdenv.mkDerivation rec {
       translators of the Bible, and have a growing collection of many hundred
       texts in around 100 languages.
     '';
-    license = licenses.gpl2;
-    maintainers = with maintainers; [ AndersonTorres ];
-    platforms = platforms.unix;
+    license = lib.licenses.gpl2;
```
Comment (AndersonTorres) [inline pkgs/by-name/sw/sword/package.nix:60]:
```suggestion
    license = lib.licenses.gpl2Plus;
```

File: pkgs/by-name/sw/sword/package.nix [hunk 4]
```diff
@@ -1,32 +1,53 @@
...
 
+stdenv.mkDerivation (finalAttrs: {
   pname = "sword";
-  version = "1.8.1";
+  version = "1.9.0";
 
   src = fetchurl {
-    url = "https://www.crosswire.org/ftpmirror/pub/sword/source/v1.8/${pname}-${version}.tar.gz";
-    sha256 = "14syphc47g6svkbg018nrsgq4z6hid1zydax243g8dx747vsi6nf";
+    url = "https://www.crosswire.org/ftpmirror/pub/sword/source/v${lib.versions.majorMinor finalAttrs.version}/sword-${finalAttrs.version}.tar.gz";
```
Comment (AndersonTorres) [inline pkgs/by-name/sw/sword/package.nix:16]: IIRC it has a GitHub repo. I will look at it later

Comment (afh) [reply pkgs/by-name/sw/sword/package.nix:16]: Is it [bibletime/crosswire-sword-mirror](https://github.com/bibletime/crosswire-sword-mirror)?

Comment (afh) [reply pkgs/by-name/sw/sword/package.nix:16]: Upon close inspection that ☝️ repo does not have releases or tags, which would be nice to have to build the package…

Comment (AndersonTorres) [reply pkgs/by-name/sw/sword/package.nix:16]: Hum, it looks a bit harder.

http://www.crosswire.org/sword/develop/index.jsp

They use their own SVN repo instead.

Comment (afh) [reply pkgs/by-name/sw/sword/package.nix:16]: @AndersonTorres maybe you could lend your expertise on this:

When I change the package `src` to be:
```
  src = fetchsvn {
    url = "https://crosswire.org/svn/sword/tags/sword-${lib.replaceStrings ["."] ["-"] finalAttrs.version}";
    hash = "sha256-QkCc894vrxEIUj4sWsB0XSH57SpceO2HjuncwwNCa4o=";
  };
```

and do a `nix build .#sword` I see the following error:
```
       > svn: E155000: Destination directory exists; please remove the directory or use --force to overwrite
       > svn: E155000: '/nix/store/r001qz6gcg5ks5crlksa577xwkdhg6fq-sword-sword-1-9-0-rHEAD/bindings/objc/sword' already exists
```

What could be the reason for that? Any ideas?

Comment (AndersonTorres) [reply pkgs/by-name/sw/sword/package.nix:16]: Why not to use url = https://crosswire.org/svn/sword and rev=sword-1.9.0?

Comment (afh) [reply pkgs/by-name/sw/sword/package.nix:16]: Because it fails with:
> svn: E205000: Syntax error in revision argument 'sword-1.9.0'

Comment (AndersonTorres) [reply pkgs/by-name/sw/sword/package.nix:16]: I will investigate it later. For now the expression is OK.

**Review by AndersonTorres — APPROVED** [review]

**Comment by afh** [comment]
Thanks for merging, @Aleksanaa, much appreciated.

@AndersonTorres do let me know if I can help in regards to the SVN `src.url`, happy to assist.

**Pull request merged**